src/tileset_writer.rs

```diff
@@ -0,0 +1,158 @@
+use std::fs::{File, create_dir_all, remove_dir_all};
+use std::io::prelude::*;
+use std::path::Path;
+
+use json::JsonValue;
+
+use crate::octrees::OctNode;
+use crate::pnts_writer::PntsWriter;
+use crate::glb_writer::GlbWriter;
+
+#[derive(Clone, PartialEq)]
+pub enum TileType {
+    Pnts,
+    Glb,
+}
+
+impl TileType {
+    pub fn get_extension(&self) -> &str {
+        match self {
+            Self::Pnts => "pnts",
+            Self::Glb => "glb"
+        }
+    }
+}
+
+pub struct TilesetWriter {
+    tile_type: TileType
+}
+
+impl TilesetWriter {
+    pub fn new(tile_type: TileType) -> Self {
+        Self {
+            tile_type
+        }
+    }
+
+    pub fn save(&self, dirname: &str, root: &OctNode) {
+        if Path::new(dirname).exists() {
+            remove_dir_all(dirname)
+                .expect("could not remove old tileset directory");
+        }
+        create_dir_all(dirname).expect("could not create tileset directory");
+        println!("Generating tileset JSON...");
+        self.make_tileset_json(dirname, root);
+        println!("Generating .pnts files...");
+        self.make_contents(dirname, root);
+    }
+
+    /// Generate a tileset.json file by traversing the tree and collecting
+    /// data
+    ///
+    /// See https://github.com/CesiumGS/3d-tiles/tree/master/specification#reference-tileset
+    fn make_tileset_json(&self, dirname: &str, root: &OctNode) {
+        let prefix = "0";
+        let root_tile = self.make_tileset_json_recursive(root, &prefix);
+        let tileset = object!{
```

Comment:
Also add a `3DTILES_metadata` extension to include other data:

- number of iterations
- number of points per initial set
- number of initial set copies
- number of transforms in each IFS
- total number of points
- which algorithm was used
- which plotter was used
- node capacity